### PR TITLE
Cleaned up old solutions, added new ones

### PR DESCRIPTION
**Pull request by Groders**

### CODE REVIEW

File: ctci/chapter1/1.2.py [hunk 1]
```diff
@@ -5,13 +5,14 @@
 #strip whitespace
 #check len
 #check same number of chars 
-def isPermutation(s1: str, s2: str):
+def isPermutation(s1: str, s2: str) -> bool:
```
Comment (kwojcicki) [inline ctci/chapter1/1.2.py:8]: :fireworks: python3

File: ctci/chapter1/1.4.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #1.4
 #Palindrome Permutation
 
-def permutPalindrome(s: str):
+def permutPalindrome(s: str) -> bool:
```
Comment (kwojcicki) [inline ctci/chapter1/1.4.py]: nit: permute

File: ctci/chapter1/1.8.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#an element in an MxN matrix is 0, its entire row and column are set to O.
+
+def zeroMatrix(m):
+    if (len(m) == 0 or len(m[0]) == 0):
+        return m
+
+    h = [False] * len(m)
+    w = [False] * len(m[0])
+
+    for i, col in enumerate (m):
```
Comment (kwojcicki) [inline ctci/chapter1/1.8.py:12]: https://stackoverflow.com/a/16548714/12165177 seems promising to clean this up

File: ctci/chapter1/1.9.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#1.9
+#String Rotation: Assume you have amethod isSubstring which checks 
+#ifone word is asubstring of another. Given two strings, s1 and s2, 
+#write code to check if s2 is a rotation of s1 using only one call to 
+#isSubstring (e.g.,"waterbottle"is a rotation of"erbottlewat").
+
+#in keyword is bascially substring
+def isUnisSubstring(s1: str, s2: str) -> bool:
```
Comment (kwojcicki) [inline ctci/chapter1/1.9.py]: Is there a typo in the method name?

Comment (mkuang97) [reply ctci/chapter1/1.9.py]: I believe this should be isRotation() is what he meant?

File: ctci/chapter1/1.7.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#1.7
+#Rotate Matrix: Given an image represented by an NxN matrix, 
+#where each pixel in the image is 4 bytes, write a method to 
+#rotate the image by 90 degrees. can you do this in place?
+from typing import List
+
+#question was for bytes, but we can ignore that :)
+def rotateMatrix(m: List[List[bytes]]) -> List[List[bytes]]:
+    n = len(m)//2
```
Comment (kwojcicki) [inline ctci/chapter1/1.7.py:9]: Given you do range(n) is it necessary for you to floor this value?

File: ctci/chapter1/1.7.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    l = len(m)
```
Comment (kwojcicki) [inline ctci/chapter1/1.7.py]: probably easier to do l = len(m) - 1 given thats all you do with l

File: ctci/chapter1/1.2.py (hunk 1 above)
Comment (Groders) [reply ctci/chapter1/1.2.py:8]: :o

File: ctci/chapter1/1.7.py (hunk 5 above)
Comment (Groders) [reply ctci/chapter1/1.7.py:9]: I think range complained if i gave it a float/none floored value.

File: ctci/chapter1/1.7.py (hunk 6 above)
Comment (Groders) [reply ctci/chapter1/1.7.py]: True

File: ctci/chapter1/1.8.py (hunk 3 above)
Comment (Groders) [reply ctci/chapter1/1.8.py:12]: Ehh, I don't really like that, not really obvious what is going on

Comment (kwojcicki) [reply ctci/chapter1/1.8.py:12]: Programming wasn't obvious before I could read, just get good :)

File: ctci/chapter1/1.7.py (hunk 5 above)
Comment (kwojcicki) [reply ctci/chapter1/1.7.py:9]: fair enough.

**Comment by kwojcicki** [comment]
LGTM :sheep: :ship: it!

**Pull request merged**